Feature/isv assistant

JeanMarcMilletScality opened the pull request.

ZiyangLinScality committed:
Add ISV module components and types for Veeam and Commvault integration

ZiyangLinScality committed:
Remove unused baseSchema import and improve tooltip formatting

ZiyangLinScality committed:
Enhance Veeam configuration validation and update routing for ISV steps

ZiyangLinScality committed:
Refactor ISV configuration types and components to support dynamic bucket management and improve tooltips for better user guidance

JeanMarcMilletScality committed:
add logo for applications

JeanMarcMilletScality committed:
add ISV modal and component for it

JeanMarcMilletScality committed:
add ISV modal and component for it

ZiyangLinScality committed:
Merge branch 'refactor/ISV' into feature/isv-assistant

ZiyangLinScality committed:
Add account name type selection and radio group component for ISV configuration

JeanMarcMilletScality committed:
Update logo components : scale up or down logo to have a more consistent size between logos

JeanMarcMilletScality committed:
remove ribbon and more to come card

ZiyangLinScality committed:
Add IAM user management to ISV configuration for Veeam

JeanMarcMilletScality committed:
add bucket name field to isv configuration

JeanMarcMilletScality committed:
change routes and remove side bar from isv configuration

update card for isv, change label in configuration, add default bucket in config

JeanMarcMilletScality committed:
update modal and welcome modal

JeanMarcMilletScality committed:
test jsx element error

JeanMarcMilletScality committed:
ts error jsx element namaspace

ZiyangLinScality committed:
Add option to generate new access key/secret key in ISV configuration

JeanMarcMilletScality committed:
update Welcome modal name and import

JeanMarcMilletScality committed:
modal should not be displayed by default

JeanMarcMilletScality committed:
Improve card style to avoid shift

JeanMarcMilletScality committed:
add veeam vbo in modal choice

JeanMarcMilletScality committed:
Update modal and change open modal label in accountList

JeanMarcMilletScality committed:
change modal name in micro app configuration

ZiyangLinScality committed:
Refactor Veeam ISV configuration and application selection

ZiyangLinScality committed:
Enhance ISV configuration with IAM user management and dynamic account selection

JeanMarcMilletScality committed:
Improve bucket field and bucket validation

ZiyangLinScality committed:
Refactor BucketField component with simplified bucket management

ZiyangLinScality committed:
Refactor ISV configuration with platform-specific adjustments and UI improvements

JeanMarcMilletScality committed:
Enhance ISV card component with link support and theme integration

JeanMarcMilletScality committed:
Integrate ISV modal into account management flow with state handling

ZiyangLinScality committed:
Update Commvault ISV configuration with IAM user validation and advanced settings

JeanMarcMilletScality committed:
Add 'noopener' to window.open calls in ISVModal and WelcomeModal for security

JeanMarcMilletScality committed:
Add documentation links to ISVModal and WelcomeModal for better user guidance

JeanMarcMilletScality committed:
change isv order

ZiyangLinScality committed:
Refactor ISVApplyActions to use ISVSkipModal and update modal content with platform-specific information

ZiyangLinScality committed:
Update ISVApplyActions to destructure platform from propsConfiguration for improved clarity

ZiyangLinScality committed:
Refactor Veeam imports to use centralized constants file for improved maintainability

ZiyangLinScality committed:
Update useMutationTableData to use VeeamTableProps for improved type consistency

ZiyangLinScality committed:
Refactor ISVSteps to import useSearchParams from react-router and comment out unused selectors in Accounts tests

ZiyangLinScality committed:
Add capacity management features and refactor ISV components for improved functionality

ZiyangLinScality committed:
Fix error handling in CapacityFormSection and update Veeam validation logic for improved accuracy

ZiyangLinScality committed:
Refactor ISVConfiguration and Veeam modules to streamline validation logic and improve bucket capacity handling

ZiyangLinScality committed:
Refactor validation schemas for ISV modules and add common validator

ZiyangLinScality committed:
Add useMutationActions hook for ISV account creation workflow

JeanMarcMilletScality committed:
wip usebucketmutation in applyaction

ZiyangLinScality committed:
Refactor ISV mutation actions and components to support multi-bucket workflow

ZiyangLinScality committed:
Refactor ISV types and components to consolidate bucket and configuration types

JeanMarcMilletScality committed:
Refactor Veeam policy constants and add GET_VEEAM_POLICY function for improved bucket policy management (use same structure as GET_COMMVAULT_POLICY)

JeanMarcMilletScality committed:
Add usePolicyMutation hook for managing IAM policies for multiple buckets:
Check if policy exists then either create a new one or edit the existing one by adding buckets name in resource
also check if no more than 5 policies and delete one if true

JeanMarcMilletScality committed:
Move account check after useEffect to avoid no account page display on existing account when there is a lot of account

JeanMarcMilletScality committed:
Refactor ISV policy management by moving GET_ISV_POLICY to a new utility file and updating the Veeam policy structure for improved clarity and maintainability

JeanMarcMilletScality committed:
Update documentation links in ISVList to use relative paths

bert-e commented:
# Hello jeanmarcmilletscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko-ui/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Conflict




There is a conflict between your branch `feature/isv-assistant` and the
destination branch `development/3.0`.

Please resolve the conflict on **the feature branch** (`feature/isv-assistant`).

```sh
git fetch && \
git checkout origin/feature/isv-assistant && \
git merge origin/development/3.0
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:feature/isv-assistant
```

JeanMarcMilletScality committed:
Use new policy mutation in useMutationActions and update GET_ISV_POLICY cases to uses platform Id

JBWatenbergScality commented on src/js/mutations.ts:
```diff
@@ -165,6 +167,80 @@ const useCreatePolicyMutation = () => {
+      );
+      const policy = policies.Policies.find(
+        (policy) => policy.PolicyName === policyName,
+      );
```
Why not doing a get policy instead of listing all the policies ? If. you get a not found it means it doesn't exists.

JBWatenbergScality commented on src/js/mutations.ts:
```diff
@@ -165,6 +167,80 @@ const useCreatePolicyMutation = () => {
+            `arn:aws:s3:::${bucket}`,
+          ])
+          .flat(),
+      );
```
We might need to enforce the policy we want to push. Alsp the Statement order may not be guaranteed so we have to retreive the expected Statement to patch.

JBWatenbergScality commented on src/js/mutations.ts:
```diff
@@ -165,6 +167,80 @@ const useCreatePolicyMutation = () => {
   });
 };
 
+const usePolicyMutation = () => {
```
This name seems a bit too generic. This one is only designed to be used for your use case, I would suggest reviewing the name accordingly

ZiyangLinScality committed:
Add account prop to ISV configuration and mutation actions to support existing account workflow

ZiyangLinScality committed:
Support S3 client-based bucket creation for ISV workflow

JeanMarcMilletScality committed:
Fix attact policy to user
Add bucket tagging functionality to useMutationActions for improved resource management

ZiyangLinScality committed:
Decode URI-encoded policy document before parsing

JeanMarcMilletScality committed:
Refactor ISV configuration to include card info and improve structure for Commvault and Veeam modules

JeanMarcMilletScality committed:
Uncomment condition to display welcome modal
update cardcConfig type

ZiyangLinScality committed:
Enhance ISV workflow with access key handling and dynamic configuration

JeanMarcMilletScality committed:
Refactor ISV module imports and update ISVSteps to include Veeam, Commvault, and VeeamVBO configurations

JeanMarcMilletScality committed:
Add immutability section handling for Veeam and Commvault in ISVSummary

JeanMarcMilletScality committed:
Remove debug logging from GET_ISV_POLICY function

JeanMarcMilletScality committed:
Fix case sensitivity for Commvault in immutability section and add default return for other applications

ZiyangLinScality committed:
Refactor useMutationActions to pre-initialize mutations for improved performance and readability

JeanMarcMilletScality committed:
Refactor access key handling to support multiple keys and improve UI component styling

JeanMarcMilletScality committed:
fix typo

JeanMarcMilletScality committed:
Add loading state to account selection dropdown in ISVConfiguration component

JeanMarcMilletScality committed:
Add fixed menu position to Select component in ISVConfiguration to avoid issue with hidden overflow and change of height for advanced setting

JeanMarcMilletScality committed:
Remove banner for access key list

JeanMarcMilletScality committed:
Adjust FormSection widths in ISVConfiguration and BucketField components for improved layout

Rename renderFormGroup to renderBucketNameFormSection for clarity in BucketField component

ZiyangLinScality committed:
Refactor ISVApplyActions to support multi-mutation workflow for bucket creation and tagging

ZiyangLinScality committed:
Improve table layout and bucket labeling in ISV components

ZiyangLinScality committed:
Refactor bucket creation and tagging mutations to use S3 client directly

JeanMarcMilletScality committed:
Add banner in Summary

JeanMarcMilletScality committed:
remove comment

ZiyangLinScality committed:
Add Veeam-specific object repository mutations to ISV apply actions

JeanMarcMilletScality committed:
Move all files for ISV assistant into the same folder

JeanMarcMilletScality committed:
Update WelcomeModal path in rspack.config.ts to reflect new component location

JeanMarcMilletScality committed:
Remove custom validation for classic bucket creation

JeanMarcMilletScality committed:
Add ISV application tag handling to Overview component

JeanMarcMilletScality committed:
fix import path for useGetS3ServicePoint in VeeamSummary component

JeanMarcMilletScality committed:
Uncomment Create button and clean import

JeanMarcMilletScality committed:
Fix import path for useNextLogin in WelcomeModal test

Fix import path for useGetS3ServicePoint in VeeamSummary test

JeanMarcMilletScality committed:
Refactor import paths for VeeamConstants to ISV module structure
Create folder for Modal related components
Move Logo into ISV folder

JeanMarcMilletScality committed:
remove VeeamConstants to refer to constants

JeanMarcMilletScality committed:
Add box-sizing style to ISVSteps component to avoid Finish button being against page bottom

ZiyangLinScality committed:
Add unit tests for ISV policy generation utility

JeanMarcMilletScality committed:
Refactor ISV policy tests and utility to improve readability and maintainability

ZiyangLinScality committed:
Add unit tests for useCapacityUnit hook and related utility functions

ZiyangLinScality committed:
Add unit tests for useMultiMutation hook

ZiyangLinScality committed:
Add unit tests for useMutationActions hook

ZiyangLinScality committed:
Add unit tests for useIAMUser hook

ZiyangLinScality committed:
Add unit tests for RadioGroup component

ZiyangLinScality committed:
Add unit tests for ArtescaLogo component

ZiyangLinScality committed:
Add unit tests for ISVSteps component and context

ZiyangLinScality committed:
Enhance WelcomeModal test suite with improved rendering and mocking

JeanMarcMilletScality committed:
Refactor CardISV component by removing ManualISVCard and simplifying structure

JeanMarcMilletScality committed:
Refactor WelcomeModal tests and add ThemeProvider

JeanMarcMilletScality committed:
Add unit tests for ISVModal component

JeanMarcMilletScality committed:
Refactor CardISV and ISVModal components for improved styling and structure

JeanMarcMilletScality committed:
Move Components for Veeam bucket Overview in Veeam module and update import

JeanMarcMilletScality committed:
Remove Veeam feature

JeanMarcMilletScality committed:
Change wording when no account remove Veeam mention

JeanMarcMilletScality committed:
Update account tests to replace Veeam references with ISV and improve selector usage

JeanMarcMilletScality committed:
Remove Veeam references from CapacityFormSection component

JeanMarcMilletScality committed:
Refactor Veeam module imports and remove unused components from old Veeam assistant

JeanMarcMilletScality committed:
Add new application tag and update references in Veeam components

JeanMarcMilletScality committed:
redirect to first bucket on finish

JeanMarcMilletScality committed:
Enhance Overview component to support ISV application tags alongside Veeam, updating bucket identification logic and display.

JeanMarcMilletScality committed:
Update and improve Overview tests

Disable edition for buckets managed by Veeam

JeanMarcMilletScality committed:
Fix AccountDetails test: add waitFor and update useAuth mock

ZiyangLinScality committed:
Add comprehensive test suite for ISVConfiguration component

ZiyangLinScality committed:
Refactor ISVConfiguration tests: remove commented code and enhance form validation checks

ZiyangLinScality committed:
Clean up ISVConfiguration test imports

ZiyangLinScality committed:
Add comprehensive test suite for ISVApplyActions component

JeanMarcMilletScality committed:
Refactor ISVSummary component: move Container styled component outside of Level4FormSection and export ISVSummaryProps type
Add test for Summary

JeanMarcMilletScality committed:
Enhance AccountDetails tests: mock userData with original session state and add getToken method

JeanMarcMilletScality committed:
Refactor ISVSummary tests: rename describe block and skip clipboard copy tests

JBWatenbergScality commented on src/react/ISV/components/BucketField.tsx:
```diff
@@ -0,0 +1,230 @@
+  bucketNamePlaceholder: string;
+  bucketNumber: number;
+  bucketNameTooltip: React.JSX.Element;
+  register: UseFormRegister<FormValues>;
```
nit: instead of drill-downing the useform props you could use the `FormProvider` context from the library and get register from that

JBWatenbergScality commented on src/react/ISV/components/BucketField.tsx:
```diff
@@ -0,0 +1,230 @@
+}) => (
+  <FormGroup
+    id={`bucketName-${index}`}
+    label={bucketNumber > 1 ? `Bucket #${index + 1} name` : 'Bucket name'}
```
nit: So for multiple bucket we have inconsistencies between the first bucket label and the others ? I think it is fine but might need to double check with @Cuervino

JBWatenbergScality commented on src/react/ISV/components/BucketField.tsx:
```diff
@@ -0,0 +1,230 @@
+      }));
+      replace(updatedFields);
+    }
+  }, [platform, fields.length]);
```
nit: We might want to move that into the "onChange" handler of platform, this is discutable we can review this together later on

JBWatenbergScality commented on src/react/ISV/components/BucketField.tsx:
```diff
@@ -0,0 +1,230 @@
+      return;
+    }
+
+    setBucketNumber(newNumber);
```
I feel like bucketNumber state should be derived from the buckets length in the formhook state

JBWatenbergScality commented on src/react/ISV/components/BucketField.tsx:
```diff
@@ -0,0 +1,230 @@
+    setBucketNumber(newNumber);
+
+    if (newNumber < fields.length) {
+      for (let i = fields.length - 1; i >= newNumber; i--) {
+        remove(i);
+      }
```
nit: you could just simply slice the array to the desired index and do a replace

JBWatenbergScality commented on src/react/ISV/components/BucketField.tsx:
```diff
@@ -0,0 +1,230 @@
+  const bucketNamePlaceholder = `${platform}-bucket-name`;
+
+  const xCoreLibrary = useXCoreLibrary();
+  const { useClusterCapacity } =
+    xCoreLibrary === XCORE_NOT_AVAILABLE ? () => ({}) : xCoreLibrary;
```
When XCORE is unavailable I think that this will raise an error because you cannot destructurate `useClusterCapacity` from `() => ({})`

JBWatenbergScality commented on src/react/ISV/components/ISVApplyActions.tsx:
```diff
@@ -0,0 +1,271 @@
+  onMutationReady: (key: string, mutation: MutationWithKey) => void;
+}) => {
+  const putVeeamFolderMutation = usePutObjectMutation();
+  const putVeeamSystemXmlMutation = usePutObjectMutation();
+  const putVeeamCapacityXmlMutation = usePutObjectMutation();
+
+  useMemo(() => {
+    onMutationReady(`putVeeamFolder-${bucket.name}`, {
+      ...putVeeamFolderMutation,
+      key: `putVeeamFolder-${bucket.name}`,
+    });
+    onMutationReady(`putVeeamSystemXml-${bucket.name}`, {
+      ...putVeeamSystemXmlMutation,
+      key: `putVeeamSystemXml-${bucket.name}`,
+    });
+    onMutationReady(`putVeeamCapacityXml-${bucket.name}`, {
+      ...putVeeamCapacityXmlMutation,
+      key: `putVeeamCapacityXml-${bucket.name}`,
+    });
```
nit nit nit: I wonder if this Veeam specificity should be injected using some kind of reverse control from the config file of ISV

ZiyangLinScality committed:
Add navigation tests for ISVConfiguration component

ZiyangLinScality committed:
Fix BucketField: Remove unnecessary state and simplify bucket number handling

JBWatenbergScality commented on src/react/ISV/components/ISVApplyActions.tsx:
```diff
@@ -0,0 +1,271 @@
+    onMutationReady(`createBucket-${bucket.name}`, {
+      ...createBucketMutation,
+      key: `createBucket-${bucket.name}`,
+    });
```
nit: onMutationReady caller could take the first argument and add it as a key automatically.

ZiyangLinScality committed:
Simplify BucketField: Replace remove method with replace for bucket array management

JBWatenbergScality commented on src/react/ISV/components/ISVApplyActions.tsx:
```diff
@@ -0,0 +1,271 @@
+        style={{ width: '50rem' }}
+      >
+        <div style={{ height: '32rem' }}>
+          <Table>
```
Why not using core-ui table here ?

ZiyangLinScality committed:
Refactor BucketField: Separate initial and platform-specific field updates

JBWatenbergScality commented on src/react/ISV/components/ISVCapacityFormSection.tsx:
```diff
@@ -0,0 +1,138 @@
+  const { setValue } = useFormContext();
+  const { capacityValue, capacityUnit } = useCapacityUnit(
+    clusterCapacityStatus === 'success'
+      ? clusterCapacity * (0.8 / index + 1)
```
Shouldn't it be the total number of buckets here instead of an index ?
```suggestion
      ? clusterCapacity * (0.8 / (index + 1))
```

JBWatenbergScality commented on src/react/ISV/components/ISVCapacityFormSection.tsx:
```diff
@@ -0,0 +1,138 @@
+    }
+  }, [clusterCapacityStatus]);
+
+  return <CapacityFormSection index={index} />;
```
autoFocusEnabled props is not forwarded, is it expected ?

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+const NameField = ({
+  register,
+  control,
+  errors,
```
Those props could be retrieved from the form hook context instead

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+        required
+        labelHelpTooltip={
+          platform.fieldOverrides.find(
+            (field) => field.name === FORM_FIELDS.ACCOUNT_NAME,
```
I think it's missing a condition here for the IAM User tooltip

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+  fieldType,
+  getIAMUsersMutation = null,
+  setAccount = null,
+}) => {
```
Missing typing for the props of this component

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+  status,
+  options,
+  platform,
+  type,
```
Should be typed form IAMUserTypeOptions[number].value

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+                autoComplete="off"
+                placeholder={
+                  status === 'success' && options.length !== 0
+                    ? `${platform.id}-backup`
```
Why do we hardcode `-backup` here ?

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+  setAccount = null,
+}) => {
+  const isAccount = fieldType === 'account';
+  const fieldName = isAccount
```
It seems that we have some duplica in the props in between fieldType and `getIAMUsersMutation` because we won't have (I assume) `getIAMUsersMutation` in the account case, similarly with the setAccount props that shouldn't be there of the IAM Users case, right ? You could have some props like this (var name to review): `nameMutation: ({setAccount: TYPE} | {getIAMUsersMutation})` and then derive `isAccount` and `fieldType` from that.

JBWatenbergScality replied:
`setAccount ` and `getIAMUsersMutation ` could also just be handlers such as `onFieldNameChange` to delegate this to the parent making. The labels and tooltip could also be props, in this way this component could be natively reused for other use cases than Account and IAM user later on

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+    IAMUserNameType,
+    onIAMUsersLoaded: (users) => {
+      if (users.length > 0) {
+        methods.setValue(FORM_FIELDS.IAM_USER_NAME_TYPE, 'existing');
```
nit: 
```suggestion
        formMethods.setValue(FORM_FIELDS.IAM_USER_NAME_TYPE, 'existing');
```

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+  });
+
+  const onSubmit = async (data: ISVConfig) => {
+    console.log('Form submitted with data:', data);
```
```suggestion
```

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+
+  const onSubmit = async (data: ISVConfig) => {
+    console.log('Form submitted with data:', data);
+    setConfig(data);
```
What is the goal of this setConfig call/prop ?

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+    });
+  };
+
+  const formRef = useRef(null);
```
I'm unsure about the use of this ref, is it really useful ?

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+  return (
+    <FormProvider {...methods}>
+      <ISVSkipModal
+        isOpen={skip}
```
nit:

```suggestion
        isOpen={skipConfirmationModalIsDisplayed}
```

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+              type="button"
+              variant="outline"
+              onClick={() => {
+                console.log(account);
```
```suggestion
```

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,477 @@
+            </Accordion>
+          )}
+
+          {platform.id === 'veeam-vbo' && renderVeeamApplication()}
```
This should also be rendered for veeam-vbr I believe

JeanMarcMilletScality replied:
I think its the name that is misleading: it's a field to choose between the version of Microsoft Office v8+ or v7-

ZiyangLinScality committed:
Add constants for bucket number range in BucketField

JBWatenbergScality commented on src/react/ISV/components/ISVSteps.tsx:
```diff
@@ -0,0 +1,112 @@
+  const [searchParams] = useSearchParams();
+  const id = searchParams.get('platform');
+
+  const getApplication = () => {
```
This could be a field of platform so that when we add support for anew app, in theory we can only change the isvlist

JBWatenbergScality commented on src/react/ISV/components/ISVSummary.tsx:
```diff
@@ -0,0 +1,336 @@
+  height: ${spacing.r32};
+`;
+
+const getImmutabilitySectionInfo = (
```
This should be in the ISVList to centralise customisations.

ZiyangLinScality committed:
Refactor BucketField: Remove explicit register prop and use useFormContext for registration

JBWatenbergScality commented on src/react/ISV/components/Modal/ISVModal.test.tsx:
```diff
@@ -0,0 +1,112 @@
+    await waitFor(() => {
+      expect(secondCard).toBeChecked();
+    });
+    //? don't know why but right is going to the left card
+    await userEvent.keyboard('{arrowright}');
```
Can probably be fixed thanks to a tab-index prop or reviewing the labels

JeanMarcMilletScality replied:
Happen only in tests, in ui works as expected

JBWatenbergScality commented on src/react/ISV/components/Modal/ISVModal.tsx:
```diff
@@ -0,0 +1,212 @@
+  return (
+    <Stack direction="vertical" gap="r16">
+      <Text isEmphazed variant="Large">
+        Which application would you like to configure with your ARTESCA?
```
`ARTESCA` shouldn't be hard coded anywhere in Zenko, Zenko can be used in other contexts. Maybe we can merge a first version as is and think about this one in next PR

JBWatenbergScality commented on src/react/ISV/components/Modal/WelcomeModal.tsx:
```diff
@@ -0,0 +1,192 @@
+  });
+  const isZeroAccountCreated = status === 'success' && accounts.length === 0;
+  const isAlreadyInConfigurationView =
+    window.location.pathname.endsWith('/configuration');
```
Why not using useLocation hook from react router ?

JBWatenbergScality commented on src/react/ISV/components/Modal/WelcomeModal.tsx:
```diff
@@ -0,0 +1,192 @@
+          <Text variant="Large">Welcome to</Text>
+          <ArtescaLogo />
+          <Text variant="Large">ARTESCA - Connector Marketplace</Text>
+          <Text variant="Large" color="textSecondary">
+            Draft Concept
+          </Text>
```
@Cuervino can we remove this "Draft concept" now ?

Cuervino replied:
Yes of course.

JBWatenbergScality commented on src/react/ISV/components/RadioGroup.tsx:
```diff
@@ -0,0 +1,98 @@
+            disabled={disabled}
+          />
+          <RadioContent>
+            <RadioLabel>{option.label}</RadioLabel>
```
Aren't we missing a html-for here ? Could also be a label instead of a span

JBWatenbergScality commented on src/react/ISV/components/__tests__/ISVApplyActions.test.tsx:
```diff
@@ -0,0 +1,303 @@
+import { render, screen, fireEvent } from '@testing-library/react';
```
For this test, I think we mock a bit too many things making the unit test too tightly coupled with the current implementation. I'm ok to approve this first iteration and review this after

JBWatenbergScality commented on src/react/ISV/components/__tests__/ISVConfiguration.test.tsx:
```diff
@@ -0,0 +1,446 @@
+      expect(screen.getByText('Exit Veeam assistant?')).toBeInTheDocument();
+    });
+
+    // it('should navigate to accounts page when confirming skip', async () => {
+    //   renderComponent();
+
+    //   // Click the skip button
+    //   await userEvent.click(screen.getByText('Skip Use case configuration'));
+
+    //   // Click the "Exit" button in the modal
+    //   await userEvent.click(
+    //     screen.getByRole('button', { name: 'Exit configuration' }),
+    //   );
+
+    //   await waitFor(() => {
+    //     expect(mockNavigate).toHaveBeenCalledWith('/accounts');
+    //   });
+    // });
+
```
This test is commented out

JBWatenbergScality commented on src/react/ISV/components/__tests__/ISVConfiguration.test.tsx:
```diff
@@ -0,0 +1,446 @@
+
+    it('should close modal without navigating when canceling skip', async () => {
+      // Reset mocks before test
+      mockNavigate.mockClear();
```
Should be done in a befroeEach, tests have to remains independant from each others

JBWatenbergScality commented on src/react/ISV/components/__tests__/ISVSteps.test.tsx:
```diff
@@ -0,0 +1,209 @@
+  useSearchParams: jest.fn(),
+}));
+
+// Mock Scality core-ui components
+jest.mock('@scality/core-ui', () => ({
+  spacing: { r16: '16px' },
+  Stepper: ({ steps }) => (
+    <div data-testid="stepper">
+      {steps.map((step, index) => (
+        <div key={index} data-testid={`step-${index}`}>
+          {step.label}
+        </div>
+      ))}
+    </div>
+  ),
+  Text: ({ children }) => <div>{children}</div>,
+}));
+
+jest.mock('@scality/core-ui/dist/next', () => ({
+  Box: ({ children }) => <div data-testid="box">{children}</div>,
+}));
+
```
Core-ui shouldn't be mocked out. Mocking with data-testid is a bad practice that I discourage because it coupled the unit test with the current implementation.

ZiyangLinScality committed:
Add comprehensive test suite for BucketField component

JBWatenbergScality commented on src/react/ISV/components/__tests__/ISVSummary.test.tsx:
```diff
@@ -0,0 +1,435 @@
+    expect(selectors.bucketSection()).toBeInTheDocument();
+    expect(selectors.certificateSection()).toBeInTheDocument();
+    // // Verify the copy buttons
+    // await user.click(selectors.copyServicePointButton());
+    // await expect(navigator.clipboard.readText()).resolves.toBe(SERVICE_POINT);
+    // await user.click(selectors.copyAccessKeyButton());
+    // await expect(navigator.clipboard.readText()).resolves.toBe(ACCESS_KEY);
+    // await user.click(selectors.copySecretKeyButton());
+    // await expect(navigator.clipboard.readText()).resolves.toBe(SECRET_KEY);
+    // await user.click(selectors.copyBucketNameButton());
+    // await expect(navigator.clipboard.readText()).resolves.toBe(BUCKET_NAME);
+    // await user.click(selectors.copyRegionButton());
+    // await expect(navigator.clipboard.readText()).resolves.toBe(DEFAULT_REGION);
+    // await user.click(selectors.copyAllButton());
+    // await expect(navigator.clipboard.readText()).resolves.toBe(
+    //   `Service point\t${SERVICE_POINT}\nRegion\t${DEFAULT_REGION}\nAccess key ID\t${ACCESS_KEY}\nSecret Access key\t${SECRET_KEY}\nBuckets name\t${BUCKET_NAME}`,
+    // );
```
THis test is commented out

JBWatenbergScality commented on src/react/ISV/components/__tests__/ISVSummary.test.tsx:
```diff
@@ -0,0 +1,435 @@
+    expect(selectors.bucketSection()).toBeInTheDocument();
+    expect(screen.queryAllByText(/bucket-name/i)).toHaveLength(2);
+  });
+  it.skip('should copy all information to the clipboard when clicking on the copy all button in single bucket case', async () => {
```
THis test is skipped

JBWatenbergScality commented on src/react/ISV/components/__tests__/ISVSummary.test.tsx:
```diff
@@ -0,0 +1,435 @@
+      `Service point\t${SERVICE_POINT}\nRegion\t${DEFAULT_REGION}\nAccess key ID\t${ACCESS_KEY}\nSecret Access key\t${SECRET_KEY}\nBuckets name\t${BUCKET_NAME}`,
+    );
+  });
+  it.skip('should copy all information to the clipboard when clicking on the copy all button in multiple bucket case', async () => {});
+  it.skip('should redirect to the first bucket overview page when clicking on the finish button', async () => {
```
Skipped tests

JBWatenbergScality commented on src/react/ISV/components/__tests__/RadioGroup.test.tsx:
```diff
@@ -0,0 +1,87 @@
+  it('should render all radio options', () => {
+    renderRadioGroup();
+
+    expect(screen.getByText('Option 1')).toBeInTheDocument();
```
I would recomend using a `getByRole('radio', {name: /Option 1/i})` instead

JBWatenbergScality commented on src/react/ISV/hooks/useIAMUser.tsx:
```diff
@@ -0,0 +1,111 @@
+    }
+  };
+
+  useEffect(() => {
+    if (IAMUserNameType === 'existing' && IAMUserName) {
+      checkUserAccessKeys(IAMUserName);
+    }
+  }, [IAMUserName, IAMUserNameType]);
```
This promise should be called within a useQuery hook in order to handle failure status, loading status and component unmount while the promise is ongoing safely.

JBWatenbergScality commented on src/react/ISV/hooks/useIAMUser.tsx:
```diff
@@ -0,0 +1,111 @@
+        }),
+      );
+      setUsers(mappedUsers);
+      console.log('IAM users:', mappedUsers);
```
```suggestion
```

JBWatenbergScality commented on src/react/ISV/hooks/useIAMUser.tsx:
```diff
@@ -0,0 +1,111 @@
+      );
+      setUsers(mappedUsers);
+      console.log('IAM users:', mappedUsers);
+      setStatus('success');
```
Can be derived from the mutationStatus, no need of an additional state here.

JBWatenbergScality commented on src/react/ISV/hooks/useIAMUser.tsx:
```diff
@@ -0,0 +1,111 @@
+      }
+    },
+    onError: (error) => {
+      setStatus('error');
```
Can be derived from the mutationStatus, no need of an additional state here.

JBWatenbergScality commented on src/react/ISV/hooks/useMutationActions.test.ts:
```diff
@@ -0,0 +1,786 @@
+        useMutationActions(props as any, mockBucketMutations),
+      );
+
+      console.debug('Loading test results:', result.current.data);
```
```suggestion
```

JBWatenbergScality commented on src/react/ISV/hooks/useMutationActions.test.ts:
```diff
@@ -0,0 +1,786 @@
+        useMutationActions(props as any, mockBucketMutations),
+      );
+
+      console.debug('Access key test results:', {
+        provided: props.accessKey,
+        returned: result.current,
+      });
```
```suggestion
```

JBWatenbergScality commented on src/react/ISV/utils/ISVPolicy.ts:
```diff
@@ -0,0 +1,89 @@
+      return GET_VEEAM_POLICY(buckets, enableImmutableBackup);
+    case 'commvault':
+      return GET_COMMVAULT_POLICY(buckets, enableImmutableBackup);
+    default:
```
Should be part of the ISV Config

JBWatenbergScality commented on src/react/account/AccountList.tsx:
```diff
@@ -132,14 +137,13 @@ function AccountList({ accounts }: { accounts: Account[] }) {
-                  />
-                )}
+                <Button
+                  label="Start Configuration for ISV"
```
Not sure that displaying ISV is the correct terminology that we want. CC @Cuervino

JBWatenbergScality replied:
We might want to replace this by Connector or something like this

Cuervino replied:
I propose "Start ISV Connector".

JBWatenbergScality commented on src/react/account/Accounts.tsx:
```diff
@@ -68,35 +64,31 @@ const Accounts = () => {
+                buttonSection={
+                  <>
+                    <Button
+                      label="Start Configuration for ISV"
```
Same here about the ISV labeling

JBWatenbergScality commented on src/react/ui-elements/MiddleEllipsis.tsx:
```diff
@@ -12,6 +12,8 @@ const MiddleEllipsisContainer = styled.div`
 `;
 const MiddleEllipsisText = styled.span<{ calculationDone?: boolean }>`
   opacity: ${(props) => (props.calculationDone ? '1' : '0')};
+  display: 'flex';
```
Why having touched this component ? ConstrainedText from core-ui wasn't fitting the need ?

ZiyangLinScality committed:
Refactor ISVConfiguration: Use useFormContext in NameField component

JeanMarcMilletScality committed:
Refactor ISVSummary and Hide components: Export getImmutabilitySectionInfo, update application constants, and replace MiddleEllipsis with ConstrainedText for credential display

JeanMarcMilletScality committed:
Put functions to get isv specific infos in isv config: getPolicy (to create isv policy) and immutabilitySummaryOverride (to get infos for immutable section in ISV Summary)
Add getPolicy method to IAMClient

JeanMarcMilletScality committed:
Move ISV list at the root of ISV folder

JeanMarcMilletScality committed:
use useLocation in WelcomeModal

JeanMarcMilletScality committed:
update import and variable in tests

ZiyangLinScality committed:
Refactor ISVConfiguration: Extract NameField component and simplify props

ZiyangLinScality committed:
Refactor ISVConfiguration component: rename state variable for clarity and update modal handling logic

ZiyangLinScality committed:
Refactor ISVConfiguration: Rename form methods variable for improved readability

ZiyangLinScality committed:
Update ISV Connector button label across components

ZiyangLinScality committed:
Refactor ISV Stepper Context: Remove config state and setConfig method

ZiyangLinScality committed:
Update ISVSummary test to remove config state and add application prop

ZiyangLinScality committed:
Refactor useIAMUser hook: Replace effect with useQuery for access keys management

ZiyangLinScality committed:
Update RadioGroup test to use more accessible role-based queries

JeanMarcMilletScality committed:
remove draft concept from welcome modal

JeanMarcMilletScality committed:
Refactor policy mutation logic: enforce policy actions for new buckets

ZiyangLinScality committed:
Update test mocks and assertions for ISV components

ZiyangLinScality committed:
Simplify WelcomeModal and ISVConfiguration test setup and assertions

ZiyangLinScality committed:
Refactor useMultiMutation: Simplify mutation type and handling

ZiyangLinScality committed:
Update BucketField and CapacityFormWithXcore to pass bucket number for capacity calculation

JeanMarcMilletScality committed:
Refactor ISVModal tests: update keyboard navigation
Update ISVPolicy tests : import actions from policy file

JeanMarcMilletScality committed:
Add condition to add tooltip for IAM User field

ZiyangLinScality committed:
Simplify BucketField initialization and platform handling

JeanMarcMilletScality committed:
Refactor ISVConfiguration: Use optional chaining for tooltip and label retrieval

ZiyangLinScality committed:
Update BucketField test to include default bucket initialization

ZiyangLinScality committed:
Update NameField type to support multiple account type options

ZiyangLinScality committed:
Enhance RadioGroup accessibility with explicit label and ID associations

ZiyangLinScality committed:
Update build-push-tests.yml

ZiyangLinScality committed:
Refactor ISVApplyActions tests to improve clarity

JeanMarcMilletScality committed:
remove getApplication from ISVConfig
remove all direct ref to specific isv

JeanMarcMilletScality committed:
Merge branch 'development/3.0' into feature/isv-assistant

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @monpote

* @hervedombya

* @jbwatenbergscality

* @chengyanjin

JBWatenbergScality commented on src/js/mutations.ts:
```diff
@@ -165,6 +167,110 @@ const useCreatePolicyMutation = () => {
+              `arn:aws:s3:::${bucket}/*`,
+              `arn:aws:s3:::${bucket}`,
+            ])
+            .flat(),
```
nit:

```suggestion
            .flatMap((bucket) => [
              `arn:aws:s3:::${bucket}/*`,
              `arn:aws:s3:::${bucket}`,
            ]),
```

JBWatenbergScality commented on src/react/ISV/components/ISVSteps.tsx:
```diff
@@ -0,0 +1,86 @@
+  const [searchParams] = useSearchParams();
+  const id = searchParams.get('platform');
+
+  const isvModules: ISVPlatformConfig[] = [Veeam, Commvault, VeeamVBO];
```
Can't we derive this from a config attribute from the ISV list ?  It is a bit hidden that we need to update this component when implementing a new ISV.

JBWatenbergScality commented on src/react/ISV/components/ISVSummary.tsx:
```diff
@@ -0,0 +1,298 @@
+  });
+
+  const textToCopy = `Service point\t${s3ServicePoint}\nRegion\t${DEFAULT_REGION}\n${
+    secretKey ? 'Access key ID' : 'Access key IDs'
```
is it normal to check on secretKey here instead of access key?

JBWatenbergScality commented on src/react/ISV/hooks/useIAMUser.tsx:
```diff
@@ -0,0 +1,103 @@
+  const [users, setUsers] = useState<IAMUser[]>([]);
+  const [accessKeys, setAccessKeys] = useState<string[] | null>(null);
+
+  useQuery(
```
Better use the full object approach with queryKey and queryFn because react-query 5+ deprecated this way of definining a query I think

JBWatenbergScality commented on src/react/ISV/hooks/useIAMUser.tsx:
```diff
@@ -0,0 +1,103 @@
+      enabled: IAMUserNameType === 'existing' && Boolean(IAMUserName),
+      onSuccess: (data) => {
+        if (data.activeKeys.length > 0) {
+          setAccessKeys(data.activeKeys.map((key) => key.AccessKeyId));
```
no need to define a state, useQuery give it back as the data arg

JBWatenbergScality commented on src/react/ISV/hooks/useIAMUser.tsx:
```diff
@@ -0,0 +1,103 @@
+  onShouldGenerateKey?: (shouldGenerate: boolean) => void;
+}
+
+export const useIAMUser = ({
```
I think we should review this hook all together @ziyang-lin-404 @JeanMarcMilletScality

JBWatenbergScality commented on src/react/ui-elements/Hide.tsx:
```diff
@@ -22,15 +24,7 @@ export function HideCredential({ credentials }: { credentials: string }) {
     <HideContainer>
       <HideValue shown={shown}>
-        {shown ? (
-          <MiddleEllipsis
```
I middelEllipsis still in use in the codebase ? Maybe it can be rmeoved ?

JeanMarcMilletScality replied:
Yes it is still in use

ZiyangLinScality committed:
Refactor bucket resource generation using flatMap

ZiyangLinScality committed:
Refactor useIAMUser hook to simplify state management and improve data handling

JeanMarcMilletScality committed:
Fix access key ID condition in ISVSummary component

ZiyangLinScality committed:
Remove tags from IAM user test cases

ZiyangLinScality committed:
Centralize ISV modules configuration in index file

ZiyangLinScality committed:
Refactor ISV modules validation with common validator

ZiyangLinScality committed:
Replace commonValidator with specific validation schemas in ISV modules

ZiyangLinScality committed:
Enhance IAM user configuration with dynamic form handling and key generation

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @monpote

* @chengyanjin

* @jbwatenbergscality

* @hervedombya

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,303 @@
+  const { next } = useStepper(ISVStepsIndexes.Configuration);
+  const [account, setAccount] = useState<Account | null>(null);
+
+  if (!platform.id) {
+    return null;
+  }
```
This early return is breaking the rule of hook

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,303 @@
+                    name={FORM_FIELDS.GENERATE_KEY}
+                    control={control}
+                    render={({ field: { onChange, value } }) => {
+                      console.log('value', value);
```
```suggestion
```

JBWatenbergScality commented on src/react/ISV/components/ISVConfiguration.tsx:
```diff
@@ -0,0 +1,303 @@
+
+          {accountNameType === 'existing' && accountName && (
+            <Accordion title="Advanced settings" id="advanced-settings">
+              <NameField
```
nit: should be renamed

ZiyangLinScality committed:
Add CreateOrSelectNameField component for flexible account and IAM user selection

ZiyangLinScality committed:
Remove console log from ISVConfiguration component

JBWatenbergScality reviewed (approved):
LGTM !!!!
![celebrate-hardly-art-gif-by-tacocat](https://github.com/user-attachments/assets/b5459408-c239-4c7f-a1e5-655e3e558fed)

JeanMarcMilletScality commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/3.0`



The following branches have **NOT** changed:

* `development/1.4`
* `development/1.5`
* `development/1.6`
* `development/2.0`
* `development/2.1`
* `development/2.2`



Please check the status of the associated issue None.

Goodbye jeanmarcmilletscality.




*The following options are set:* **approve**
